src/providers/limit-provider.js

```diff
@@ -0,0 +1,114 @@
+/* CSS classes for Yamcs parameter monitoring result values. */
+const MONITORING_RESULT_CSS = {
+    'WATCH': 'is-limit--yellow',
+    'WARNING': 'is-limit--yellow',
+    'DISTRESS': 'is-limit--red',
+    'CRITICAL': 'is-limit--red',
+    'SEVERE': 'is-limit--red'
+}
+
+/* CSS classes for Yamcs range condition values. */
+const RANGE_CONDITION_CSS = {
+    'LOW': 'is-limit--lwr',
+    'HIGH': 'is-limit--upr'
+}
+
+export default class LimitProvider {
+    getLimitEvaluator(domainObject) {
+        const self = this;
+
+        return {
+            /*
+             * Evaluates a telemetry point for limit violations.
+             *
+             * Parameters:
+             *     datum: the telemetry point data from the historical or realtime
+             *            plugin
+             *     valueMetadata: metadata about the telemetry point
+             *
+             * Returns:
+             *     an object with CSS class information, a violation name, and
+             *     range information, if there is a limit violates, or nothing,
+             *     if the value is within limits.
+             *
+             * The datum parameter may include the following information:
+             *     monitoringResult: the Yamcs limit monitoring result, if any
+             *     rangeCondition: the Yamcs range condition (LOW/HIGH), if any
+             *     alarmRange: an array of alarm ranges for different monitoring
+             *         results, or omitted, if no alarm ranges are defined
+             */
+            evaluate: function (datum, valueMetadata) {
+                if (valueMetadata && datum.monitoringResult
+                        && datum.monitoringResult in MONITORING_RESULT_CSS) {
+                    let evaluationResult;
+
+                    if (datum.rangeCondition
+                            && datum.rangeCondition in RANGE_CONDITION_CSS) {
+                        evaluationResult = self.getLimitRange(datum, datum.monitoringResult, valueMetadata);
+                    }
+
+                    return evaluationResult;
+                }
+            }
+        };
+    }
+
+    /*
+     * Adds limit range information to an object based on the monitoring
+     * result.
+     *
+     * Parameters:
+     *     datum: the telemetry point data from the historical or realtime
+     *            plugin
+     *     result: the monitoring result information from Yamcs
+     *     valueMetadata: metadata about the telemetry point
+     *
+     * Returns: the object that is to be the result of the
+     *          limit evaluation
+     */
+    getLimitRange(datum, result, valueMetadata) {
+        const key = valueMetadata && valueMetadata.key;
+        if (key === undefined) {
```

Comment:
Not a huge deal, but if you declare range and key together you can do an OR check and have one empty return.

Reply:
yeah had that before, but wanted to return asap 😅